var/spack/repos/builtin/packages/quantlib/package.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Quantlib(CMakePackage):
+    """The QuantLib project is aimed at providing a comprehensive software
+    framework for quantitative finance. QuantLib is a free/open-source library
+    for modeling, trading, and risk management in real-life."""
+
+    homepage = "https://www.quantlib.org/"
+    url = "https://github.com/lballabio/QuantLib/releases/download/v1.31.1/QuantLib-1.31.1.tar.gz"
+
+    maintainers("TheQueasle")
+
+    version("1.31.1", sha256="13b5346217153ae3c185e0c640cc523a1a6522c3a721698b2c255fd9a1a15a68")
+
+    variant("benchmark", default=True, description="Build benchmark")
+    variant("examples", default=True, description="Build examples")
+    variant("test", default=True, description="Build test suite")
+    variant("openmp", default=False, description="Detect and use OpenMP")
+    variant("punit", default=False, description="Enable the parallel unit test runner")
+    variant(
+        "sessions",
+        default=False,
+        description="Singletons return different instances for different sessions",
+    )
+    variant("threadsafe", default=False, description="Enable the thread-safe observer pattern")
+    variant("tracing", default=False, description="Tracing messages should be allowed")
+    variant(
+        "errorfunctions",
+        default=False,
+        description="Error messages should include current function information",
+    )
+    variant(
+        "errorlines",
+        default=False,
+        description="Error messages should include file and line information",
+    )
+    variant("extra", default=False, description="Extra safety checks should be performed")
+    variant("highres", default=False, description="Enable date resolution down to microseconds")
+    variant(
+        "null",
+        default=False,
+        description="Enable the implementation of Null as template functions",
+    )
+    variant("tagged", default=False, description="Library names use layout tags")
+    variant("tidy", default=False, description="Use clang-tidy when building")
+    variant("indexed", default=False, description="Use indexed coupons instead of par coupons")
+    variant("std", default=True, description="Enable all QL_USE_STD_* options")
+
+    depends_on("boost")
+
+    def cmake_args(self):
+        args = []
+
+        if "+benchmark" in self.spec:
+            args += ["-DQL_BUILD_BENCHMARK=ON", "-DQL_INSTALL_BENCHMARK=ON"]
+        else:
+            args += ["-DQL_BUILD_BENCHMARK=OFF", "-DQL_INSTALL_BENCHMARK=OFF"]
```

Comment:
These constructs should be shortened to:
```suggestion
        args = [
            self.define_from_variant("QL_BUILD_BENCHMARK", "benchmark"),
            self.define_from_variant("QL_INSTALL_BENCHMARK", "benchmark"),
        ]
```
See e.g. https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake for